jetcd-launcher/src/main/java/io/etcd/jetcd/launcher/EtcdClusterFactory.java

```
@@ -64,6 +72,9 @@ public void start() {
         } catch (InterruptedException e) {
           throw new RuntimeException(e);
         }
+        if (failedToStart.get()) {
+          throw new IllegalStateException("Cluster failed to start");
```

we should not swallow the exception that has caused the failure